clients/extension/scripts/popup.js

```diff
@@ -150,160 +59,132 @@ function createOptionItem(text, id, isChecked) {
-  statusMsg.textContent = 'Starting...';
-
-  showRunningSubpage();
-
-  const feedbackEl = find('.feedback');
-  feedbackEl.textContent = '';
-  const {selectedCategories, useDevTools} = settings;
-  /** @type {LH.Flags} */
-  const flags = {throttlingMethod: useDevTools ? 'devtools' : 'simulate'};
-
-  try {
-    await background.runLighthouseInExtension(flags, selectedCategories);
-
-    // Close popup once report is opened in a new tab
-    window.close();
-  } catch (err) {
-    let message = err.friendlyMessage || err.message;
-    let includeReportLink = true;
-
-    // Check for errors in how the user ran Lighthouse and replace with a more
-    // helpful message (and remove 'Report Error' link).
-    for (const [test, replacement] of Object.entries(NON_BUG_ERROR_MESSAGES)) {
-      if (err.message.includes(test) ||
-          (err.friendlyMessage && err.friendlyMessage.includes(test))) {
-        message = replacement;
-        includeReportLink = false;
-        break;
-      }
-    }
-
-    feedbackEl.textContent = message;
-
-    if (includeReportLink) {
-      feedbackEl.className = 'feedback feedback-error';
-      feedbackEl.appendChild(buildErrorCopyButton(message, err));
-    }
-
-    hideRunningSubpage();
-    background.console.error(err);
-  }
-
-  isRunning = false;
+  url.searchParams.append('utm_source', 'lh-chrome-ext');
+  window.open(url.href);
 }
 
 /**
  * Generates a document fragment containing a list of checkboxes and labels
  * for the categories.
- * @param {BackgroundPage} background Reference to the extension's background page.
- * @param {Array<string>} selectedCategories
+ * @param {Settings} settings
  */
-function generateOptionsList(background, selectedCategories) {
+function generateOptionsList(settings) {
   const frag = document.createDocumentFragment();
 
-  background.getDefaultCategories().forEach(category => {
-    const isChecked = selectedCategories.includes(category.id);
+  ExtensionController.DEFAULT_CATEGORIES.forEach(category => {
+    const isChecked = settings.selectedCategories.includes(category.id);
     frag.appendChild(createOptionItem(category.title, category.id, isChecked));
   });
 
-  const optionsList = find('.options__list');
-  optionsList.appendChild(frag);
+  const optionsCategoriesList = find('.options__categories');
+  optionsCategoriesList.appendChild(frag);
 }
 
-/**
- * Initializes the popup's state and UI elements.
- */
-async function initPopup() {
-  chrome.tabs.query({active: true, lastFocusedWindow: true}, function(tabs) {
-    if (tabs.length === 0) {
-      return;
-    }
+function fillDevToolsShortcut() {
+  const el = find('.devtools-shortcut');
+  const isMac = /mac/i.test(navigator.platform);
+  el.textContent = isMac ? '⌘⌥I (Cmd+Opt+I)' : 'F12';
+}
 
-    siteURL = new URL(tabs[0].url || '');
+function logVersion() {
+  // @ts-ignore: Run when in extension context, but not in unit tests.
+  if (typeof window.ControllerMock !== 'undefined') return;
```

Comment:
mocking the global `chrome` would also make this more robust

Reply:
done https://github.com/GoogleChrome/lighthouse/pull/9193/commits/df9cd1301cd3e5093f15eb11ebd0b6d0d28e1ecf